### PR TITLE
Create, Update, Get, Delete Task Assignment

### PR DESCRIPTION
**Pull request by marek-vysoky**
It is dependent on PR:
https://github.com/box/box-windows-sdk-v2/pull/224

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @marek-vysoky has signed the CLA. Thanks for the pull request!

File: Box.V2/Config/BoxConfig.cs [hunk 1]
```diff
@@ -98,6 +98,15 @@ public BoxConfig(string clientId, string clientSecret, Uri redirectUri)
...
         public virtual Uri SharedItemsUri { get { return new Uri(BoxApiUri, Constants.SharedItemsString); } }
         /// <summary>
+        /// Gets the task assignments endpoint URI.
+        /// </summary>
+        /// <value>
+        /// The task assignments endpoint URI.
+        /// </value>
+        public virtual Uri TaskAssignmentsEndpointUri { get { return new Uri(BoxApiUri, Constants.TaskAssignmentsString); } }
+
+        public virtual Uri TasksEndpointUri { get { return new Uri(BoxApiUri, Constants.TasksString); } }
```
Comment (coolboy) [inline Box.V2/Config/BoxConfig.cs]: done

File: Box.V2/Config/IBoxConfig.cs [hunk 2]
```diff
@@ -58,6 +58,15 @@ public interface IBoxConfig
...
         Uri SharedItemsUri { get; }
         /// <summary>
+        /// Gets the task assignments endpoint URI.
+        /// </summary>
+        /// <value>
+        /// The task assignments endpoint URI.
+        /// </value>
+        Uri TaskAssignmentsEndpointUri { get; }
+
+        Uri TasksEndpointUri { get; }
```
Comment (coolboy) [inline Box.V2/Config/IBoxConfig.cs]: done

File: Box.V2/Managers/BoxTasksManager.cs [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+    public class BoxTasksManager : BoxResourceManager
+    {
+        public BoxTasksManager(IBoxConfig config, IBoxService service, IBoxConverter converter, IAuthRepository auth, string asUser = null, bool? suppressNotifications = null)
+            : base(config, service, converter, auth, asUser, suppressNotifications) { }
+
+        /// <summary>
+        /// Used to assign a task to a single user. There can be multiple assignments on a given task.
+        /// </summary>
+        /// <param name="taskAssignmentRequest">The task assignment request.
+        /// taskAssignmentRequest.Task.Id - Id of the task,
```
Comment (coolboy) [inline Box.V2/Managers/BoxTasksManager.cs]: done

File: Box.V2/Managers/BoxTasksManager.cs [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+
+            IBoxResponse<BoxTaskAssignment> response = await ToResponseAsync<BoxTaskAssignment>(request).ConfigureAwait(false);
+
+            return response.ResponseObject;
+        }
+
+        /// <summary>
+        /// Used to update a task assignment.
+        /// </summary>
+        /// <param name="taskAssignmentUpdateRequest">The task assignment update request.
```
Comment (coolboy) [inline Box.V2/Managers/BoxTasksManager.cs]: done

File: Box.V2/Managers/BoxTasksManager.cs [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+
+            IBoxResponse<BoxTaskAssignment> response = await ToResponseAsync<BoxTaskAssignment>(request).ConfigureAwait(false);
+
+            return response.Status == ResponseStatus.Success;
+        }
+
+        /// <summary>
+        /// Used to create a single task for single user on a single file.
+        /// </summary>
+        /// <param name="taskCreateRequest">The task create request.
```
Comment (coolboy) [inline Box.V2/Managers/BoxTasksManager.cs]: done

File: Box.V2/Managers/BoxTasksManager.cs [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+            IBoxResponse<BoxTask> response = await ToResponseAsync<BoxTask>(request).ConfigureAwait(false);
+
+            return response.ResponseObject;
+        }
+
+        /// <summary>
+        /// Updates a specific task.
+        /// </summary>
+        /// <param name="taskId">Id of the task.</param>
+        /// <param name="taskUpdateRequest">The task update request.
```
Comment (coolboy) [inline Box.V2/Managers/BoxTasksManager.cs]: done

File: Box.V2/Managers/BoxTasksManager.cs [hunk 7]
```diff
@@ -0,0 +1,211 @@
...
+                .Method(RequestMethod.Post)
+                .Payload(_converter.Serialize(taskUpdateRequest));
+
+            IBoxResponse<BoxTask> response = await ToResponseAsync<BoxTask>(request).ConfigureAwait(false);
+
+            return response.ResponseObject;
+        }
+
+        /// <summary>
+        /// Permanently deletes a specific task..
```
Comment (coolboy) [inline Box.V2/Managers/BoxTasksManager.cs]: done

File: Box.V2/Models/Request/BoxTaskAssignmentRequest.cs [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+    public class BoxTaskAssignmentRequest 
+    {
+        /// <summary>
+        /// The task this assignment is for.
+        /// </summary>
+        [JsonProperty(PropertyName = "task")]
+        public BoxTaskRequest Task { get; set; }
+
+        /// <summary>
+        /// The user this assignment is for. .
```
Comment (coolboy) [inline Box.V2/Models/Request/BoxTaskAssignmentRequest.cs]: done

File: Box.V2/Models/BoxTaskAssignment.cs [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Box.V2.Models
+{
+    /// <summary>
+    ///  Assignment for a given task
+    /// </summary>
+    public class BoxTaskAssignment : BoxEntity
```
Comment (coolboy) [inline Box.V2/Models/BoxTaskAssignment.cs]: done

**Pull request merged**